moves/moves.js

```diff
@@ -63,13 +78,35 @@ window.Moves = (function() {
     function setupRoleChangeListener() {
         const roleSelect = document.getElementById('role');
         if (roleSelect) {
-            roleSelect.addEventListener('change', (event) => {
-                const selectedRole = event.target.value;
-                renderMovesForRole(selectedRole);
-                
-                // Restore any learned moves from the URL
-                restoreLearnedMoves(selectedRole);
-            });
+        roleSelect.addEventListener('change', async (event) => {
+            const selectedRole = event.target.value;
+            
+            // Render cards first and wait for completion
+            if (selectedRole && window.Cards) {
```

Comment:
Feels like this is maybe duplication?  Should be one render loop from scratch and changes should all trigger that function